gslib/utils/user_agent_helper.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains helper for appending user agent information."""
+
+import sys
+import gslib
+from gslib.utils import system_util
+from gslib.utils.arg_helper import GetArgumentsAndOptions
+
+
+def GetUserAgent(metrics_off=True):
+  """Using the command arguments return a suffix for the UserAgent string.
+
+  Args:
+    metrics_off: boolean, whether the MetricsCollector is disabled.
+
+  Returns:
+    str, A string value that can be appended to an existing UserAgent.
+  """
+  _, opts = GetArgumentsAndOptions()
```

Comment:
If we create a module level constant to hold the UserAgent info then perhaps this call will not be necessary. You would be able to pass the args. e.g on line 57 of `gslib.__main__.py`
```
opts, args = GetArgumentsAndOptions()
gslib.USER_AGENT = GetUserAgent(metrics.MetricsCollector.IsDisabled())
boto.UserAgent += gslib.USER_AGENT
```

Reply:
I like it! I had originally placed the GetArgumentsAndOptions call there but then just figured it made more sense to make it a helper and run it twice, but storing the user agent there makes a lot of sense.